includes/Checker/Runtime_Environment_Setup.php

```diff
@@ -27,11 +27,27 @@ public function set_up() {
 		// Get the existing active plugins.
 		$active_plugins = get_option( 'active_plugins' );
 
+		// Get the existing permalink structure.
+		$permalink_structure = get_option( 'permalink_structure' );
+
 		// Set the new prefix.
 		$old_prefix = $wpdb->set_prefix( $table_prefix . 'pc_' );
 
 		// Create and populate the test database tables if they do not exist.
 		if ( $wpdb->posts !== $wpdb->get_var( $wpdb->prepare( 'SHOW TABLES LIKE %s', $wpdb->posts ) ) ) {
+			/*
+			 * Set the same permalink structure *before* install finishes,
+			 * so that wp_install_maybe_enable_pretty_permalinks() does not flush rewrite rules.
+			 *
+			 * See https://github.com/WordPress/plugin-check/issues/330
+			 */
+			add_action(
+				'populate_options',
+				static function () use( $permalink_structure ) {
```

Comment:
Micro nit:

```suggestion
				static function () use ( $permalink_structure ) {
```

There is always a space after `use` in the core codebase.

Reply:
Hmm this kinda stuff should be handled by WPCS :/

Reply:
Also required by WPCS 3+ I think.

Reply:
Aren‘t we on WPCS 3 already?

Reply:
Yes:

https://github.com/WordPress/plugin-check/blob/35eee0bbbceaeb5bbd7281e185d7b65d3f03c7ef/composer.json#L10

Reply:
I've reported this upstream: https://github.com/WordPress/WordPress-Coding-Standards/issues/764#issuecomment-1851018615

Reply:
Already PR opened to fix! https://github.com/WordPress/WordPress-Coding-Standards/pull/2415